### PR TITLE
ML - Sentiment Analysis - Project 2

### PR DESCRIPTION
**Pull request by Meg1456**
I have isolated the bug. Still working on fixing it.

### CODE REVIEW

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -89,13 +94,21 @@ public void storeMessage(Message message) {
...
         String idString = entity.getKey().getName();
+        System.out.println("successfully read the message entity ID");
         UUID id = UUID.fromString(idString);
         String user = (String) entity.getProperty("user");
+        System.out.println("successfully read the message entity USER");
         String text = (String) entity.getProperty("text");
+        System.out.println("successfully read the message entity TEXT");
         long timestamp = (long) entity.getProperty("timestamp");
+        System.out.println("successfully read the message entity TIMESTAMP");
+        Float sentimentScore = (Float) entity.getProperty("sentimentScore");
```
Comment (Meg1456) [inline src/main/java/com/google/codeu/data/Datastore.java]: The error seems to be in this line. This is the message I see "[INFO] java.lang.ClassCastException: java.lang.Double cannot be cast to java.lang.Float". Does anyone know a fix around this?

Comment (Reine0017) [reply src/main/java/com/google/codeu/data/Datastore.java]: I think it means that entity.getProperty("sentimentScore") returns a value of type double and you can't just cast it to float with a (float). Is it a must to use type Float? Is it possible to just use double instead?

Comment (Reine0017) [inline src/main/java/com/google/codeu/data/Datastore.java]: Will this work?

```suggestion
        Double sentimentScore = entity.getProperty("sentimentScore");
```

Comment (Reine0017) [reply src/main/java/com/google/codeu/data/Datastore.java]: Its very odd that its returning float value when you instantiated it as float below...hmm

Comment (Reine0017) [reply src/main/java/com/google/codeu/data/Datastore.java]: OR maybe if it already returns as a float you dont have to cast it? Try this link to test and see what value the google cloud api returns you: https://medium.com/google-cloud/sentiment-analysis-using-google-cloud-machine-learning-552be9b9c39b and then debug from there?

Comment (Meg1456) [reply src/main/java/com/google/codeu/data/Datastore.java]: Thank you! Ill check out the link :)) 

So if I leave it as a double here, then I just control how many decimals are printed later using .2f or the equivalent of that?

**Comment by Meg1456** [comment]
I have fixed the bug! Please review again :)

**Review by jiangyue12392 — APPROVED** [review]
Looks good to me~

**Review by Reine0017 — CHANGES_REQUESTED** [review]
There is a merging conflict (MessageServlet.java) that I think has got to do with the variable out of scope error.

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 2]
```diff
@@ -78,9 +81,17 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
 
-    Message message = new Message(user, text);
-    datastore.storeMessage(message);
+    //Perform sentiment analysis when the user submits a message
+    Document doc = Document.newBuilder()
+        .setContent(text).setType(Document.Type.PLAIN_TEXT).build();
+    LanguageServiceClient languageService = LanguageServiceClient.create();
+    Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+    double score = sentiment.getScore();
+    //System.out.println("Sentiment Score = " + score);
```
Comment (potay) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: deadcode

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 3]
```diff
@@ -78,9 +81,17 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String user = userService.getCurrentUser().getEmail();
     String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
 
-    Message message = new Message(user, text);
-    datastore.storeMessage(message);
+    //Perform sentiment analysis when the user submits a message
```
Comment (potay) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: space after //. here and elsewhere
```suggestion
    // Perform sentiment analysis when the user submits a message
```

File: src/main/webapp/js/user-page-loader.js [hunk 4]
```diff
@@ -76,7 +76,7 @@ function buildMessageDiv(message) {
   const headerDiv = document.createElement('div');
   headerDiv.classList.add('message-header');
   headerDiv.appendChild(document.createTextNode(
-      message.user + ' - ' + new Date(message.timestamp)));
+      message.user + ' - ' + new Date(message.timestamp) + ' - Sentiment Score: ' + message.sentimentScore.toFixed(2)));
```
Comment (potay) [inline src/main/webapp/js/user-page-loader.js]:
```suggestion
      `${message.user} - ${new Date(message.timestamp)} - Sentiment Score: ${message.sentimentScore.toFixed(2)}`));
```

File: src/main/java/com/google/codeu/servlets/MessageServlet.java (hunk 2 above)
Comment (Meg1456) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java]: I am using this final score obtained to generate the message in line 93 though. I don't think I fully understand what you mean bu deadcode. Can you please explain this a little more?

Comment (potay) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java]: Oh deadcode means that its lines of code not used. For example, this line has been commented out, so it should be removed.

Comment (Meg1456) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java]: Ohh, ok thanks!

**Review by Reine0017 — APPROVED** [review]
Looks great Meg!

**Pull request merged**